control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -407,7 +407,13 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		// TODO: refactor
 		topicConfig, _, err := r.topicConfig(logger, broker)
 		if err != nil {
-			return fmt.Errorf("failed to resolve broker config: %w", err)
+
+			// On finalize, we fail to get a valid config, we can safely ignore and return nil
+			if strings.Contains(err.Error(), "validating topic config") {
+				return nil
```

Comment:
I think in this case we shouldn't return immediately but remove the finalizer on the secret.

The secret is reachable (compared to https://github.com/knative-sandbox/eventing-kafka-broker/pull/2829) and we are able to remove the finalizer.

In https://github.com/knative-sandbox/eventing-kafka-broker/pull/2829, the secret was not reachable and we were not able to remove the finalizer and it was ok to just return immediately.

Reply:
you mean, the configmap, containing bogus stuff, instead of real kafka config (we are not touching secrets here)

Reply:
So, I think this code in this line should still run: https://github.com/knative-sandbox/eventing-kafka-broker/blob/9e166c9458ed7502c686b35fb541bd9d06ca0901/control-plane/pkg/reconciler/broker/broker.go#L453

```
r.removeFinalizerSecret(ctx, finalizerSecret(broker), secret)
```


That is, as you said, when the topic config in the configmap is incorrect.

However, there's nothing preventing us removing the finalizer from the secret.

Reply:
@aliok I updated a comment: basically w/ invalid config a secret is not touched etc.